src/main/java/io/jenkins/plugins/fontawesome/api/FontAwesomeIcons.java

```diff
@@ -0,0 +1,112 @@
+package io.jenkins.plugins.fontawesome.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Utility to work with icons provided by the font-awesome-api-plugin.
+ *
+ * @author strangelookingnerd
+ */
+public final class FontAwesomeIcons {
+
+    private static final Logger LOGGER = Logger.getLogger(FontAwesomeIcons.class.getName());
+    private static final String SVG_FILE_ENDING = ".svg";
+    private static final String IMAGES_SYMBOLS_PATH = "images/symbols/";
+    private static final String FONT_AWESOME_API_PLUGIN = "font-awesome-api";
+    private static final String ICON_CLASS_NAME_PATTERN = "symbol-%s plugin-" + FONT_AWESOME_API_PLUGIN;
+
+    private static final FontAwesomeIcons INSTANCE = new FontAwesomeIcons();
+
+    private final Map<String, String> availableIcons = new ConcurrentSkipListMap<>();
```

Comment:
Why is this not the faster `HashMap`?

Reply:
Can we avoid to store a map in memory here and just return it on demand?

Reply:
Was looking for a sorted, thread-safe implementation. When we are no longer keeping it in-memory, I'd opt with an `LinkedHashMap` to preserve the insertion-order. This way I can get the natural order of the keys (= names of the icons) which produces "reproducible" results.